lua/packer.lua

```diff
@@ -77,10 +61,39 @@ local config_defaults = {
   profile = {enable = false}
 }
 
+--- Initialize global namespace for use for callbacks and other data generated whilst packer is
+--- running
+_G._packer = _G._packer or {}
+
 local config = vim.tbl_extend('force', {}, config_defaults)
 local plugins = nil
+local plugin_specifications = nil
 local rocks = nil
 
+local configurable_modules = {
+  clean = false,
+  compile = false,
+  display = false,
+  handlers = false,
+  install = false,
+  plugin_types = false,
+  plugin_utils = false,
+  update = false,
+  luarocks = false,
+  log = false
+}
+local function require_and_configure(module_name)
+  local fully_qualified_name = 'packer.' .. module_name
+  local module = require(fully_qualified_name)
+  if configurable_modules[module_name] == true then return module end
```

Comment:
Why explicitly check `true` and `false` here rather that `if configurable_modules[module_name] then` etc. ? are you trying specifically to avoid `nil`?

Reply:
Yes - this is largely for future-proofing, as only some modules are configurable, and we only want to configure a module once, not every time we load it. Thus, `false` for "configurable and not configured yet", `true` for "configurable and already configured", and `nil` for "not configurable". I wanted to protect against future uses of this function accidentally trying to configure a module that shouldn't be.

That said, as long as it's clearly named `require_and_configure` and isn't overriding `require` in this file, it's probably unnecessary for now. I think I'll remove the `configurable_modules` check.